### DISCUSSION TITLE
How do I get the body parameter?

### DISCUSSION THREAD

**Discussion by myyrakle** [actix/actix-web#1860] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/16988115/103268211-7f369c80-49f6-11eb-8fb6-943dd2107250.png)
I've been trying to take a body parameter in the currently commented form. 
However, even though the format was correct, `Failed to deserialize Json from payload. Request path: /signup` error occurred.
  
So, to analyze the cause, I received it as Bytes. So the body didn't come at all.
![image](https://user-images.githubusercontent.com/16988115/103268445-ccb30980-49f6-11eb-8ca8-3961ed34850d.png)
  
In the above case, I made a request to the vscode extension rest client. 
When I made a request from the browser, it worked the same way. 
Apparently it seems to have sent the body, but it was not received.
  
What am I doing wrong?
  
The versions used are as follows.
```
[dependencies]
actix-web = "3.3.2"
actix-rt = "1"
actix-files = "0.4"
actix-service = "1.0.5"
actix-cors = "0.5.3"
log = "0.4.8"
log4rs = "0.13.0"
futures = "0.3.5"
listenfd = "0.3"
env_logger = "0.7"
diesel = { version= "1", features = ["postgres"] }
dotenv = "0.9.0"
serde_json = "1"
serde = { version = "1", features = ["derive"] }
jsonwebtoken = "7.2.0"
epoch-timestamp = "1.0.0"
```

**Comment by robjtede** [comment, marked as answer]
You've got the right handler set up. Just to be sure can you use a verified client like cURL or Postman. (The lack of a content-length request header is sus and I couldn't say whether that client inserts one.)

**Reply by myyrakle** [reply]
thank you!
